fvm/environment/accounts_status.go

```diff
@@ -70,27 +82,90 @@ func (a *AccountStatus) ToBytes() []byte {
 
 // AccountStatusFromBytes constructs an AccountStatus from the given byte slice
 func AccountStatusFromBytes(inp []byte) (*AccountStatus, error) {
-	var as AccountStatus
+	sizeChange := int64(0)
+
+	// this is to migrate old account status to new account status on the fly
+	// TODO: remove this whole block after Crescendo, when a full migration will be made.
+	if len(inp) == accountStatusSizeV1 {
+		// migrate v1 to v2
+		inp2 := make([]byte, accountStatusSizeV2)
 
-	if len(inp) == oldAccountStatusSize {
 		// pad the input with zeros
-		// this is to migrate old account status to new account status on the fly
-		// TODO: remove this whole block after v0.32.0, when a full migration will
-		// be made.
-		sizeIncrease := uint64(accountStatusSize - oldAccountStatusSize)
+		sizeIncrease := int64(accountStatusSizeV2 - accountStatusSizeV1)
 
 		// But we also need to fix the storage used by the appropriate size because
 		// the storage used is part of the account status itself.
-		copy(as[:], inp)
-		used := as.StorageUsed()
-		as.SetStorageUsed(used + sizeIncrease)
-		return &as, nil
+		copy(inp2, inp)
+		sizeChange = sizeIncrease
+
+		inp = inp2
 	}
 
-	if len(inp) != accountStatusSize {
+	// this is to migrate old account status to new account status on the fly
+	// TODO: remove this whole block after Crescendo, when a full migration will be made.
+	if len(inp) == accountStatusSizeV2 {
+		// migrate v2 to v3
+
+		inp2 := make([]byte, accountStatusSizeV2)
+		copy(inp2, inp)
```

Comment:
is this `copy()` needed? I think the append on line 134 is copying the data a second time.

Reply:
yes it. Slicing `inp` in place caused issues. I added a comment